lib/src/folding/foldable_block.dart

```diff
@@ -55,38 +55,102 @@ class FoldableBlock extends InclusiveRange with EquatableMixin {
 
     return false;
   }
+
+  bool isSameLines(FoldableBlock other) {
+    return firstLine == other.firstLine && lastLine == other.lastLine;
+  }
+
+  FoldableBlock join(FoldableBlock other) {
+    return FoldableBlock(
+      firstLine: min(firstLine, other.firstLine),
+      lastLine: max(lastLine, other.lastLine),
+      type: _getJoinType(this, other),
+    );
+  }
+
+  static FoldableBlockType _getJoinType(FoldableBlock a, FoldableBlock b) {
+    if (a.type == FoldableBlockType.imports) {
+      return FoldableBlockType.imports;
+    }
+
+    if (b.type == FoldableBlockType.imports) {
+      return FoldableBlockType.imports;
+    }
+
+    return FoldableBlockType.union;
+  }
```

Comment:
```suggestion
  static FoldableBlockType _getJoinType(FoldableBlock a, FoldableBlock b) {
    final containsImports = a.isImports || b.isImports;
    return containsImports ? FoldableBlockType.imports : FoldableBlockType.union;
  }
```

Reply:
Now imagine that we need to preserve the type if the blocks are of the same type. With this code, we would have to change lines, and also `?:` must be removed.

Those cut-offs are deliberate. They allow a lot of extension by just adding lines.